### DISCUSSION TITLE
Debugging Tools like Unity Profiler?

### DISCUSSION THREAD

**Discussion by rrtt2323** [godotengine/godot-proposals#4484] [Editor]
Now performance debug tool cannot locate the location of the stack.
Also do not support debugging for the binding language.
For example, godot-rust.
In the team are not conducive to take over the function of others.
I feel these are restricting I moved from the unity platform, the greatest obstacle.

For example,
Profiler.BeginSample("xxx");
Profiler.EndSample();

**Comment by Calinou** [comment]
This is already being tracked in https://github.com/godotengine/godot-proposals/issues/3361.

**Reply by rrtt2323** [reply]
OMG, Callgrind can be used to debug.
But the new problem is...
Godot tool chain is so long.
I think I need to learn for a long time.
